dual_net.py

```diff
@@ -173,10 +196,139 @@ def train(self, boards, pi, z):
         should match input_shape and action_size as set during initialization.
         returns the batch loss
         """
+        self.boards = [state_to_board(board) for board in boards]
```

Comment:
Okay so this line seems like it makes the net only work on the chess boards (and not extend to TicTacToe, for example). Perhaps store the env as an instance variable and let the env take care of these conversions?